src/coreclr/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -91,8 +91,8 @@ COOP_PINVOKE_HELPER(int32_t, RhGetModuleFileName, (HANDLE moduleHandle, _Out_ co
 
 COOP_PINVOKE_HELPER(void, RhpCopyContextFromExInfo, (void * pOSContext, int32_t cbOSContext, PAL_LIMITED_CONTEXT * pPalContext))
 {
-    UNREFERENCED_PARAMETER(cbOSContext);
     ASSERT((size_t)cbOSContext >= sizeof(CONTEXT));
+    memset(pOSContext, 0, cbOSContext);
     CONTEXT* pContext = (CONTEXT *)pOSContext;
```

Comment:
I am wondering whether we need to set `ContextFlags` here as well to indicate which registers are valid.

Reply:
That's a good point. I experimented with setting the `ContextFlags` to `CONTEXT_CONTROL | CONTEXT_INTEGER` and setting the correct values for the `CS` and `SS` segment registers. Once I did that, the call stack in Visual Studio after calling `RaiseFailFastException` directly pointed to the function that threw the unhandled exception.

Since this change is unrelated to this PR and will require a little more work to capture these segment registers, I'm going to move these change to a separate PR.

Reply:
I'll continue the work here: #81010.